core/internal/filetransfer/file_transfer_gcs.go

```diff
@@ -82,9 +82,9 @@ func (ft *GCSFileTransfer) Download(task *ReferenceArtifactDownloadTask) error {
 	)
 
 	// Parse the reference path to get the scheme, bucket, and object
-	bucketName, rootObjectName, err := parseReference(task.Reference)
+	bucketName, rootObjectName, err := parseCloudReference(task.Reference, gcsScheme)
```

Comment:
How do we handle the reference types we don't yet handle in core such as http/s? Do we select whether to send something to the core based on what we handle, so that here we should never see an `https://` reference?

Reply:
if we were to just turn on downloads in core with no changes, we'd send all reference types to core but would error out when trying to select the correct file transfer to use to download it (in `getStorageProvider` in `task_ref_art.go`). so we kind of go into `parseCloudReference` with the knowledge that this reference should be valid and parsable as one of our supported providers references

azure references actually start with `https://` (the structure is something like `https://myaccount.blob.core.windows.net/mycontainer/path/to/my/blob`) so we actually can't just rule out all http(s) paths beforehand